pkg/volume/empty_dir/quota.go

```diff
@@ -0,0 +1,128 @@
+package empty_dir
+
+import (
+	"bytes"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+
+	"github.com/golang/glog"
+	"k8s.io/kubernetes/pkg/api"
+)
+
+// An object for applying quota to an empty dir volume.
+type quotaApplicator interface {
+	// Apply the quota to the given EmptyDir path:
+	Apply(dir string, medium api.StorageMedium, pod *api.Pod) error
+}
+
+type xfsQuotaApplicator struct {
+}
+
+func (xqa *xfsQuotaApplicator) Apply(dir string, medium api.StorageMedium, pod *api.Pod, fsGroup *int64, quota int) error {
+	if medium == api.StorageMediumMemory {
+		glog.V(3).Infof("skipping quota application due to memory storage medium.")
+		return nil
+	}
+	if !xqa.isXfs(dir) {
+		glog.V(3).Infof("%s is not on an XFS filesystem, skipping quota application.")
```

Comment:
Hmm yes I suppose so, originally I was thinking it's a plugin that's in play regardless, but if there's an explicit config option to turn this on then failure to apply the quota does seem pretty bad. I'll rig this up, I think I'll skip wrapping the plugin entirely unless you have something in your config file for it, and then treat it like a proper error internally.